### ISSUE TITLE
EDI - Add versioning

### ISSUE THREAD

**Issue by bitingmidge** [bezlio/bezlio-apps#353]
Something that we will probably need to put into place for each of these EDI bezels that we haven't done yet is versioning. That way we can make future changes to the included javascript files without breaking existing installed bezels.

The way that it was done in other apps was to add a variable that contains the version number as a string and then build the include file path using that version number. For example, in the Accounts View app there is a `setConfig() `function in the bezel that sets there version number

`bezl.vars.config.version = "1.0"`

There is a `versions` folder that has subfolders with "frozen" copies of the included javascript files at that version level - one for each version (e.g. `versions/1.0`; `versions/1.1`).

Then there is a block of code like this:

```
// If the baseURL is not set above then we will determine a sane default based on the URL
if (bezl.vars.config.baseUrl == "") {
    if (window.location.href.indexOf("portal-dev.bezl.io") !== -1) {
        bezl.vars.config.baseUrl = "https://apps-dev.bezl.io";
    } else {
        bezl.vars.config.baseUrl = "https://apps.bezl.io";
    }
}

// If version numbers exist above they will also be added to the javascript include URLs
bezl.vars.config.baseJsUrl = bezl.vars.config.baseUrl
  							+ "/roles/sales-rep/accounts-view/"
  							+ (bezl.vars.config.version ? "versions/" + bezl.vars.config.version + "/" : "")
  							+ "js/"
bezl.vars.config.baseLibraryUrl = bezl.vars.config.baseUrl
  							+ "/libraries/"
							+ (bezl.vars.config.libVersion ? "versions/" + bezl.vars.config.libVersion + "/" : "")
```

That code is essentially determining whether to use the production or development version of the include file based on if you are using "portal" or "portal-dev". Then it builds the include path and stores it in the `bezl.vars.config.baseJsUrl` variable. NOTE: there is also a separate version level for library (i.e. common) includes that you may not be doing or need.

When you need to include a JS file the code would then resemble something like this:
```
require([bezl.vars.config.baseJsUrl + 'onDataChange.js'], function(functions) {
	functions.onDataChange(bezl)
});
```
While you are developing you set the `version` variable to an empty string and it will point to the top level code files. When the time comes to make a new version I then create the subfolder, copy in the `BezlExport.txt` file and the `js` folder and it's contents, and update the version number in the `setConfig()` function.

Questions/comments?

**Comment by gvanriper** [comment]
Ironically, that was my next step and I'm almost done with it. :)

**Comment by bitingmidge** [comment]
@gvanriper Awesome! I should have just been patient ;)

**Comment by bitingmidge** [comment]
Closed by pull request #358
